core/payment/src/dao/debit_note_event.rs

```diff
@@ -59,18 +62,36 @@ impl<'c> DebitNoteEventDao<'c> {
         node_id: NodeId,
         after_timestamp: Option<NaiveDateTime>,
         max_events: Option<u32>,
-        _app_session_id: Option<String>,
+        app_session_id: Option<String>,
     ) -> DbResult<Vec<DebitNoteEvent>> {
         readonly_transaction(self.pool, move |conn| {
             let mut query = dsl::pay_debit_note_event
                 .inner_join(event_type_dsl::pay_event_type)
+                .inner_join(
```

Comment:
Seems the join is not always required, can we join inside the `if Some( .. ) ..` block?

Reply:
not possible :(